devito/data/data.py

```diff
@@ -210,13 +311,40 @@ def _normalize_index(self, idx):
             if any(i is Ellipsis for i in idx):
                 # Explicitly replace the Ellipsis
                 items = (slice(None),)*(self.ndim - len(idx) + 1)
-                return idx[:idx.index(Ellipsis)] + items + idx[idx.index(Ellipsis)+1:]
+                items = idx[:idx.index(Ellipsis)] + items + idx[idx.index(Ellipsis)+1:]
             else:
-                return idx + (slice(None),)*(self.ndim - len(idx))
+                items = idx + (slice(None),)*(self.ndim - len(idx))
+            # Normalize slice steps:
+            processed = [slice(i.start, i.stop, 1) if
+                         (isinstance(i, slice) and i.step is None)
+                         else i for i in items]
+            return as_tuple(processed)
+
+    def _process_args(self, idx, val):
```

Comment:
this could be a free function (self isn't actually used). 

A docstring would help